src/app/containers/App/App.jsx

```diff
@@ -56,28 +56,41 @@ export const App = ({ routes, location, initialData, bbcOrigin, history }) => {
         route,
       } = getRouteProps(routes, location.pathname);
 
-      setState({
-        pageData: null,
-        status: null,
-        service: nextService,
-        variant: nextVariant,
-        id: nextId,
-        isAmp: nextIsAmp,
-        pageType: route.pageType,
-        loading: true,
-        error: null,
-        errorCode: null,
+      let loaderTimeout;
+      const loaderPromise = new Promise(resolve => {
+        loaderTimeout = setTimeout(resolve, 500);
+      });
+
+      loaderPromise.then(() => {
+        setState({
+          pageData: null,
+          status: null,
+          service: nextService,
+          variant: nextVariant,
+          id: nextId,
+          isAmp: nextIsAmp,
+          pageType: route.pageType,
+          loading: true,
+          error: null,
+          errorCode: null,
+        });
       });
 
       route.getInitialData(location.pathname).then(data => {
+        clearTimeout(loaderTimeout);
         shouldSetFocus.current = true;
-        setState(prevState => ({
-          ...prevState,
+        setState({
+          service: nextService,
+          variant: nextVariant,
+          id: nextId,
+          isAmp: nextIsAmp,
+          pageType: route.pageType,
           loading: false,
           pageData: path(['pageData'], data),
           status: path(['status'], data),
           error: path(['error'], data),
-        }));
+          errorCode: null,
```

Comment:
@hindsc52 I'd also like your thoughts on these changes. It looks fine to me, however I am bit skeptical about it. Thanks

Reply:
How come this now includes service, variant, id etc instead of `...prevState`

Reply:
I realised for the case where the `setState` is skipped, `...prevState` doesn't then contain `nextService, nextVariant, nextId` etc and therefore the last`setState` is called with values from the previous article. So setting them explicitly ensures we maintain those values between the two `setState` calls even if one was skipped.

Reply:
Thanks for the info @itimirichard